### DISCUSSION TITLE
Is Zod able to parse content of JSON files?

### DISCUSSION THREAD

**Discussion by jtuchel** [colinhacks/zod#2273] [Q&A] [answered]
I have a function that expects a JSON string and parses it to a given type, inferred by a Zod schema

```ts
const createConfigurationFromJson = (content: string): Configuration => {
  const rawConfiguration = JSON.parse(content);

  return configurationSchema.parse(rawConfiguration);
};
```

This function might throw if the JSON content is invalid JSON or Zod throws a parse error. Is it possible to let Zod parse the JSON string? So I can be sure there can only be a Zod error?

**Comment by JacobWeisenburger** [comment, marked as answer]
https://github.com/JacobWeisenburger/zod_utilz#stringtojson

If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger
